### ISSUE TITLE
Failed to parse multipart/form-data boundary

### ISSUE THREAD

**Issue by alexander-irbis** [hyperium/mime#54]
Generated by Firefox headers
```
Host: example.dev
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0
Accept: application/json
Accept-Language: uk,ru;q=0.8,ru-RU;q=0.6,en-US;q=0.4,en;q=0.2
Accept-Encoding: gzip, deflate
Referer: http://example.dev/blog/3
Content-Length: 664
Content-Type: multipart/form-data; boundary=---------------------------533179479757881361497406803
Cookie: SID=a9201fa6-35f6-4d87-a6b6-89c74d48d0d3
DNT: 1
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
```
Form data:
```
-----------------------------533179479757881361497406803
Content-Disposition: form-data; name="_wid"

f04f9a42-5afa-4cb6-a71c-de697c3db4e3
-----------------------------533179479757881361497406803
Content-Disposition: form-data; name="doc"

[skipped]
-----------------------------533179479757881361497406803--
```
Trace
```
            :: parse  Content-Type  Raw([b"multipart/form-data;\x20boundary=---------------------------533179479757881361497406803"])
            :: parse_header 3  Content-Type  Raw([b"multipart/form-data;\x20boundary=---------------------------533179479757881361497406803"])
            :: from_raw_str s "multipart/form-data; boundary=---------------------------533179479757881361497406803"

            :: mime::Mime::from_str s "multipart/form-data; boundary=---------------------------533179479757881361497406803"
            :: mime::parse s "multipart/form-data; boundary=---------------------------533179479757881361497406803"
            :: mime::parse type loop next Some((0, 109))    'm'
            :: mime::parse type loop next Some((1, 117))    'u'
            :: mime::parse type loop next Some((2, 108))    'l'
            :: mime::parse type loop next Some((3, 116))    't'
            :: mime::parse type loop next Some((4, 105))    'i'
            :: mime::parse type loop next Some((5, 112))    'p'
            :: mime::parse type loop next Some((6, 97))     'a'
            :: mime::parse type loop next Some((7, 114))    'r'
            :: mime::parse type loop next Some((8, 116))    't'
            :: mime::parse type loop next Some((9, 47))     '/'
            :: mime::parse sublevel loop next Some((10, 102))   'f'
            :: mime::parse sublevel loop next Some((11, 111))   'o'
            :: mime::parse sublevel loop next Some((12, 114))   'r'
            :: mime::parse sublevel loop next Some((13, 109))   'm'
            :: mime::parse sublevel loop next Some((14, 45))    '-'
            :: mime::parse sublevel loop next Some((15, 100))   'd'
            :: mime::parse sublevel loop next Some((16, 97))    'a'
            :: mime::parse sublevel loop next Some((17, 116))   't'
            :: mime::parse sublevel loop next Some((18, 97))    'a'
            :: mime::parse sublevel loop next Some((19, 59))    ';'
            :: mime::params_from_str s "; boundary=---------------------------533179479757881361497406803"
            :: mime::params_from_str name loop next Some((20, 32))  ' '
            :: mime::params_from_str name loop next Some((21, 98))  'b'
            :: mime::params_from_str name loop next Some((22, 111)) 'o'
            :: mime::params_from_str name loop next Some((23, 117)) 'u'
            :: mime::params_from_str name loop next Some((24, 110)) 'n'
            :: mime::params_from_str name loop next Some((25, 100)) 'd'
            :: mime::params_from_str name loop next Some((26, 97))  'a'
            :: mime::params_from_str name loop next Some((27, 114)) 'r'
            :: mime::params_from_str name loop next Some((28, 121)) 'y'
            :: mime::params_from_str name loop next Some((29, 61))  '='
            :: mime::params_from_str value !is_quoted loop next Some((30, 45))  '-'
            :: mime::Mime::from_str parse Err(InvalidToken)
```
https://github.com/hyperium/mime/blob/fda8cf121a1e2da06959f3185e3dd9bfdec64234/src/parse.rs#L131-L132

**Comment by seanmonstar** [comment]
It seems `Content-Type` has a different definition of valid media types, https://tools.ietf.org/html/rfc7231#section-3.1.1.1, even though the RFC for Media Types is stricter: https://tools.ietf.org/html/rfc6838#section-4.2

Yay Internet specs.

**Comment by seanmonstar** [comment]
Fixed in v0.3.2
